torchgeo/losses/ssl.py

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Loss functions self-supervised learning."""
+
+import torch
+import torch.nn.functional as F
+from torch.nn.modules import Module
+
+
+class NTXentLoss(Module):
+    """The Normalized-temperature Cross Entropy Loss (NT-Xent).
+
+    This loss is defined in `'A Simple Framework for Contrastive Learning of
+    Visual Representations' <https://arxiv.org/abs/2002.05709>`_.
+
+    .. versionadded:: 0.5
+    """
+
+    def forward(
+        self, z1: torch.Tensor, z2: torch.Tensor, t: torch.Tensor
+    ) -> torch.Tensor:
+        """Computes the NT-Xent loss.
+
+        Args:
+            z1: batch of vectors of shape (B, D).
+            z2: batch of vectors of shape (B, D).
+            t: temperature value to normalize the similarity vectors
```

Comment:
Should we support float temperature instead of requiring it to be a Tensor?

Reply:
IMO yes